### DISCUSSION TITLE
"run:" not handling conditionals properly

### DISCUSSION THREAD

**Discussion by Jodebu** [community/community#25380] [Actions]
<p>Hi, I’m having some trouble (well, big trouble) trying to set some outputs from a bash multiline script in a workflow and some help will be hugely appreciated.</p>
<p>This is the problematic step in my job:</p>
<pre><code class="lang-yaml">      # Set up variables
      - name: Set up variables
        id: vars
        shell: bash
        run: |
            if ["${{ github.event_name }}" == "push"]; then
              echo "::set-output name=DRIVE_FOLDER::${{ env.DRIVE_DEV_FOLDER }}"
            elif ["${{ github.event_name }}" == "pull_request"]; then
              echo "::set-output name=DRIVE_FOLDER::${{ env.DRIVE_TEST_FOLDER }}"
            fi
</code></pre>
<p>And this is the log I am getting (sensitive info redacted):</p>
<pre><code class="lang-nohighlight">Run if ["push" == "push"]; then
  if ["push" == "push"]; then
    echo "::set-output name=DRIVE_FOLDER::***"
  elif ["push" == "pull_request"]; then
    echo "::set-output name=DRIVE_FOLDER::***"
  fi
  shell: /bin/bash -e {0}
  env:
    UNITY_LICENSE: ***
    DRIVE_DEV_FOLDER: ***
    DRIVE_TEST_FOLDER: ***
/home/runner/work/_temp/f53e2665-c70c-404f-ad37-35543db96cc5.sh: line 1: [push: command not found
/home/runner/work/_temp/f53e2665-c70c-404f-ad37-35543db96cc5.sh: line 3: [push: command not found
</code></pre>
<p>Thank you very much in advance, please feel free to ask for anything else that could help you understand the issue.</p>

**Comment by airtower-luna** [comment, marked as answer]
<ol>
<li>You need to have spaces between square brackets and the expression. The opening bracket is actually a command (namely, an alias of the <code>test</code> command).</li>
<li>String comparison for equality in the <code>test</code> command is just a single “equals” sign.</li>
</ol>
<p>So you need to do it like this:</p>
<pre><code class="lang-bash">if [ "${{ github.event_name }}" = "push" ]; then
    echo "::set-output name=DRIVE_FOLDER::***"
elif [ "${{ github.event_name }}" = "pull_request" ]; then
    echo "::set-output name=DRIVE_FOLDER::***"
fi
</code></pre>

**Comment by Jodebu** [comment]
<p>Thank you kind sir, you brilliantly solved my problem.</p>

**Comment by airtower-luna** [comment]
<p>You’re very welcome, but it’s “ma’am” in my case. :slightly_smiling_face:</p>

**Comment by teracross** [comment]
Note for posterity - in case anyone else stumbles upon this post. The formatting of the GitHub actions set-output command has been changed and now follows format of: 
```
- name: Set output
run: echo "{name}={value}" >> $GITHUB_OUTPUT
```
source: https://github.blog/changelog/2022-10-10-github-actions-deprecating-save-state-and-set-output-commands/
